core/src/main/java/org/kiwix/kiwixmobile/core/history/HistoryAdapter.kt

```diff
@@ -0,0 +1,149 @@
+/*
+ * Kiwix Android
+ * Copyright (c) 2019 Kiwix <android.kiwix.org>
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+package org.kiwix.kiwixmobile.core.history
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ImageView
+import android.widget.TextView
+import androidx.core.content.ContextCompat
+import androidx.recyclerview.widget.RecyclerView.Adapter
+import androidx.recyclerview.widget.RecyclerView.ViewHolder
+import butterknife.BindView
+import butterknife.ButterKnife
+import org.kiwix.kiwixmobile.core.R
+import org.kiwix.kiwixmobile.core.R.drawable
+import org.kiwix.kiwixmobile.core.R.layout
+import org.kiwix.kiwixmobile.core.R.string
+import org.kiwix.kiwixmobile.core.R2.id
+import org.kiwix.kiwixmobile.core.extensions.setBitmapFromString
+import org.kiwix.kiwixmobile.core.history.HistoryListItem.DateItem
+import org.kiwix.kiwixmobile.core.history.HistoryListItem.HistoryItem
+import org.threeten.bp.LocalDate
+import org.threeten.bp.format.DateTimeFormatter
+
+internal class HistoryAdapter(
+  val historyList: List<HistoryListItem>,
+  val deleteList: List<HistoryListItem>,
+  val itemClickListener: OnItemClickListener
+) : Adapter<ViewHolder>() {
+
+  override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
+    return if (viewType == TYPE_ITEM) {
```

Comment:
when if is used as an expression it is allowed to omit the brackets

Reply:
Like this?

```
override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder =
    if (viewType == TYPE_ITEM)
      Item(parent.inflate(R.layout.item_bookmark_history, false))
    else
      Category(parent.inflate(R.layout.header_date, false))

```

Reply:
yup looks good, also rename `Item/Category` to have a ViewHolder suffix. Also move them to separate files

Reply:
If I am interpreting the bind correctly the `ItemViewHolder` is dependent on the `deleteList`. I could solve it by setting the `favicon` of the `HistoryItem` beforehand, as `favicon` is a string I think this requires some decoding of the drawable and it feels a bit ugly. Any suggestions? 

This is what the bind looks like as an inner class:
```

override fun bind(item: HistoryItem) {
      title.text = item.historyTitle
      if (deleteList.contains(item)) {
        favicon.setImageDrawableCompat(
          R.drawable.ic_check_circle_blue_24dp
        )
      } else {
        favicon.setBitmap(Base64String(item.favicon))
      }
      ...
}

```

Reply:
You can pass deleteHistory to the constructor of the ViewHolder